lib/semantics/resolve-names.cc

```diff
@@ -3812,6 +3813,9 @@ void DeclarationVisitor::Post(
     }
     if (auto *s{MakeTypeSymbol(bindingName, ProcBindingDetails{*procedure})}) {
       SetPassNameOn(*s);
+      if (GetAttrs().test(Attr::DEFERRED)) {
+        context().SetError(*s);
```

Comment:
This could be written `context().SetError(*s, GetAttrs().test(Attr::DEFERRED));`

Reply:
Sure, but that looks to me like it could *clear* a previously-set error flag if the predicate were satisfied.  Which it probably doesn't, but I think it's more clear as written.